sqlitecluster/SQLiteNode.cpp

```diff
@@ -1433,25 +1437,29 @@ void SQLiteNode::_onMESSAGE(SQLitePeer* peer, const SData& message) {
             // there's a backlog of commands, these can get stale, and by the time they reach the follower, it's already
             // behind, thus never catching up.
             if (_state == FOLLOWING) {
-                // Attach all of the state required to populate a SYNCHRONIZE_RESPONSE to this message. All of this is
-                // processed asynchronously, but that is fine, the final `SUBSCRIBE` message and its response will be
-                // processed synchronously.
-                SData request = message;
-                uint64_t count = 0;
-                string hash;
-                peer->getCommit(count, hash);
-                request["peerCommitCount"] = to_string(count);
-                request["peerHash"] = hash;
-                request["peerID"] = to_string(_getIDByPeer(peer));
-
-                // The following properties are only used to expand out our log macros.
-                request["name"] = _name;
-                request["peerName"] = peer->name;
-
-                // Create a command from this request and pass it on to the server to handle.
-                auto command = make_unique<SQLiteCommand>(move(request));
-                command->initiatingPeerID = peer->id;
-                _server.acceptCommand(move(command), true);
+                if (_gracefulShutdown()) {
+                    // This will cause the remote peer to reconnect (it will not necessarily give a fantastic error message for this), and choose a different sync peer.
+                    // This prevents us have having leftover synchronization responses in progress as we shut down.
+                    SWARN("Asked to help SYNCHRONIZE but shutting down.");
+                    SData response("SYNCHRONIZE_RESPONSE");
+                    peer->sendMessage(response);
```

Comment:
I think it would be better to add another field that gets caught in `_recvSynchronize` and makes the error message more obvious (and/or makes it an info not a warn and/or doesn't log anything) but NAB

Reply:
Adding this.